crates/pink-libs/subrpc/src/contracts/mod.rs

```diff
@@ -0,0 +1,447 @@
+use alloc::vec::Vec;
+use objects::*;
+use pink_extension::chain_extension::signing;
+use primitive_types::H256;
+use scale::{Decode, Encode};
+
+use crate::contracts::objects::{ContractCall, WeightV2};
+
+pub mod objects;
+
+pub type ContractId = [u8;32];
+pub type Balance = u128;
+
+#[derive(Encode, Decode, Debug)]
+#[repr(u8)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum Error {
+    FailedToDryRunContract(crate::Error),
+    FailedToQueryContract(crate::Error),
+    FailedToCreateTransaction(crate::Error),
+    FailedToSendTransaction(crate::Error),
+    FailedToDecode,
+    InvalidAddressLength,
+    NoResult,
+    FailedToReadResult,
+}
+
+#[derive(Encode, Decode, Debug)]
+#[repr(u8)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum ContractError {
+    Error,
+}
+
+pub type Result<T> = core::result::Result<T, Error>;
+
+pub struct InkContract<'a>  {
+    rpc: &'a str,
+    pallet_id: u8,
+    call_id: u8,
+    contract_id: &'a ContractId,
+}
+
+
+impl<'a> InkContract<'a>  {
+
+    pub fn new(rpc: &'a str, pallet_id: u8, call_id: u8, contract_id: &'a ContractId) -> Self {
+        InkContract {
+            rpc,
+            pallet_id,
+            call_id,
+            contract_id
+        }
+    }
+
+    pub fn query <A: Encode, R: Decode>(
+        &self,
+        origin: [u8; 32],
+        contract_method: [u8; 4],
+        contract_args: Option<&A>,
+        value: Balance,
+    ) -> Result<R> {
+        self.query_at(
+            origin,
+            contract_method,
+            contract_args,
+            value,
+            None
+        )
+    }
+
+    pub fn query_at <A: Encode, R: Decode>(
+        &self,
+        origin: [u8; 32],
+        contract_method: [u8; 4],
+        contract_args: Option<&A>,
+        value: Balance,
+        at: Option<H256>
+    ) -> Result<R> {
+
+        /*
+        let origin : [u8;32] = signing::get_public_key(signer, signing::SigType::Sr25519)
+            .try_into()
+            .map_err(|_| Error::InvalidAddressLength)?;
+
+         */
```

Comment:
Let's just remove the test codes.

Reply:
Indeed I forgot to remove this commented code